runtime/ftplugin/query.lua

```diff
@@ -2,5 +2,29 @@
 -- Language:	Tree-sitter query
 -- Last Change:	2022 Mar 29
 
+if vim.b.did_ftplugin == 1 then
+  return
+end
+
+-- use treesitter over syntax
+vim.treesitter.start()
+
+-- query linter
+local buf = vim.api.nvim_get_current_buf()
+local query_lint_on = vim.b[buf].query_lint_on
+  or vim.g.query_lint_on
+  or { 'BufWrite', 'InsertLeave', 'TextChanged', 'BufEnter' }
```

Comment:
```suggestion
  or { 'BufRead', 'BufWrite' }
```
(I thought we decided on that due to lagginess -- we can always add more later if performance improves.)

Reply:
I fixed the lagginess. With caching everything is fast.

Reply:
So BufRead, BufWrite, InsertLeave', 'TextChanged' ?

Reply:
I'd probably avoid `TextChanged`; this is too noisy. We default to `InsertLeave` and `CursorHoldI` for diagnostics, and I think it would make sense to align with that.

Reply:
what about `BufRead`? The query linter will often be used to debug unchanged query files after a parser change. It is strange that you need to start editing to see diagnostics.

Reply:
Yes, I think that's a good idea. LSP diagnostics are triggered by push, so that situation is different in that regard.

Reply:
Sorry to ask again. I didn't understand whether you referred to `BufEnter` or `TextChanged`. What should I set the list to?

Reply:
"No" to `TextChanged`, "Yes" to `BufEnter`.